packages/@aws-cdk/aws-lambda/lib/languages/jvm.ts

```diff
@@ -0,0 +1,84 @@
+import cdk = require('@aws-cdk/cdk');
+import fs = require('fs');
+import { join } from 'path';
+import { Function as LambdaFunction, PartialFunctionProps } from '../function';
+import { Runtime } from '../runtime';
+import { LambdaBuilderCode } from './lambda-builder';
+
+export interface JvmCodeProps {
+  path: string;
+
+  /**
+   * @default inferred from project structure (pom.xml or maven, build.gradle for gradle)
+   */
+  dependencyManager?: string;
```

Comment:
Not sure I understand what the possible values are here? Should this be an enum? Maybe we need two "Code" classes such as `MavenProject` and `GradleProject` so we can strong-type support for each dep manager?